org.adempiere.base/src/org/compiere/print/DataEngine.java

```diff
@@ -1066,12 +1068,19 @@ else if (m_group.isFunctionColumn(pdc.getAD_PrintFormatItem_ID(), functions[f]))
 										if (display.startsWith("@SCRIPT")) {
 											display = display.replace("@SCRIPT", "");
 											value = parseVariable(display, pdc, pd);
-											Interpreter bsh = new Interpreter ();
-											try {
-												value = bsh.eval(value.toString());
+											if(Objects.toString(value, "").startsWith("\"Item not found: ")) {
```

Comment:
It is very fragile to build logic depending on the exact content of an error message. If it doesn't always work for immediate evaluation, we can always defer it for all `@SCRIPT` node.

Reply:
To fix this, I moved the parsing of all `@SCRIPT` nodes after the 'all rows' loop